src/index.js

```diff
@@ -35,25 +35,102 @@ const QUERY = 'query'
 // AuthenticationRequest sending methods
 const REDIRECT = 'redirect'
 
+// Local storage keys
+const CURRENT_PROVIDER = 'solid.current-provider'
+const CURRENT_CREDENTIALS = 'solid.current-user'
+
+// Local storage key prefixes
+const RP_BY_PROVIDER = 'oidc.rp.by-provider.'
+const PROVIDER_BY_STATE = 'oidc.provider.by-state.'
+
 class ClientAuthOIDC {
   /**
    * @constructor
    * @param [options={}]
    * @param [options.window=Window] Optionally inject global browser window
    * @param [options.store=localStorage] Optionally inject localStorage
+   * @param [options.providerUri] {string} Previously selected provider uri
+   *   (typically loaded from storage)
+   * @param [options.redirectUri] {string} This app's callback redirect uri,
+   *   defaults to the current window's uri.
+   * @param [options.debug] {Function}
    */
   constructor (options = {}) {
     this.window = options.window || global.window
     this.store = options.store || global.localStorage
 
+    this.debug = options.debug || console.error.bind(console)
+
     this.currentClient = null
-    this.providerUri = null
-    this.webId = null
-    this.idToken = null
-    this.accessToken = null
+    this.providerUri = options.providerUri
+    this.redirectUri = options.redirectUri
+    this.webId = options.webId
+    this.idToken = options.idToken
+    this.accessToken = options.accessToken
     this.method = REDIRECT  // only redirect is currently supported
   }
 
+  /**
+   * Factory method, returns an auth client instance initialized with options
+   * or defaults (including loading stored credentials and current provider
+   * from local storage).
+   *
+   * @param options {Object} See constructor options
+   * @param [options.store=localStorage] {Store}
+   * @param [options.providerUri] {string}
+   *
+   * @returns {ClientAuthOIDC}
+   */
+  static from (options) {
+    let store = options.store || global.localStorage
```

Comment:
`window.localStorage` is what you want. `global` throws a ReferenceError in the browser.